coordinator/src/orderbook/mod.rs

```diff
@@ -30,120 +41,317 @@ pub mod websocket;
 #[cfg(test)]
 mod tests;
 
-pub struct Orderbook {
+struct Orderbook {
     pool: Pool<ConnectionManager<PgConnection>>,
-    shorts: Shorts,
-    longs: Longs,
+    shorts: OrderbookSide,
+    longs: OrderbookSide,
+    // TODO(holzeis): Split up order matching fee rate into taker and maker fees.
+    order_matching_fee_rate: Decimal,
 }
 
-pub struct Shorts(Vec<Order>);
+struct OrderbookSide {
+    orders: Vec<Order>,
+}
 
-impl Shorts {
-    pub fn add(&mut self, order: Order) {
-        self.0.push(order);
+impl OrderbookSide {
+    pub fn new() -> Self {
+        Self { orders: vec![] }
     }
-    pub fn remove(&mut self, order_id: Uuid) {
-        self.0.retain(|o| o.id != order_id)
+    /// adds the given order to the orderbook.
+    pub fn add_order(&mut self, order: Order) {
+        self.orders.push(order);
+        self.sort(order.direction);
     }
-    pub fn sort(&mut self) {
-        self.0.sort_by(|a, b| {
-            if a.price.cmp(&b.price) == Ordering::Equal {
-                return a.timestamp.cmp(&b.timestamp);
-            }
-            // Descending order.
-            b.price.cmp(&a.price)
-        })
+    /// updates an order in the orderbook. Ignores the update if order does not exist.
+    pub fn update_order(&mut self, updated_order: Order) {
+        if let Some(order) = self.orders.iter_mut().find(|o| o.id == updated_order.id) {
+            *order = updated_order;
+        }
+        self.sort(updated_order.direction);
```

Comment:
❓Do we need `update`? How about removing and adding a new one?

Reply:
I think its not used at the moment, but we have an `OrderbookMessage`, so I've added it preliminary.